### PR TITLE
I223 files pattern matching

### PR DESCRIPTION
**Pull request by ota-sandr**

### CODE REVIEW

File: ci/unit_tests/wawCommons/absoluteFilePaths_test.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    def test_multiplePattterns(self):
+        """Tests if mutiple patterns in absoluteFilePaths behave like there is operator OR between them."""
+        directory = os.path.join(self.dataBasePath, 'directory1')
+        pathList = list(wawCommons.absoluteFilePaths(directory, ["*.test", "*"]))
+
+        testFile1 = os.path.abspath(os.path.join(directory, "file1.test"))
+        assert testFile1 in pathList
+        testFile2 = os.path.abspath(os.path.join(directory, "fileA.ext"))
+        assert testFile2 in pathList
```
Comment (ondrejtomas) [inline ci/unit_tests/wawCommons/absoluteFilePaths_test.py]: More tests (suggestions):
- pattern that contains hierarchy, e.g. "*/*.test" 
- pattern that matches more then one file (`test_multiplePattterns` contains "*", but the first pattern already matches "file1.test"... so maybe copy this test without the first pattern)
- invalid pattern
- invalid directory
- in the first 2 tests I would use multiple files to be sure that it does not return only the first match

File: ci/unit_tests/wawCommons/getFilesAtPath_test.py [hunk 2]
```diff
@@ -4,11 +4,123 @@
...
+
+    def test_multiplePattterns(self):
+        """Tests if mutiple patterns in getFilesAtPath behave like there is operator OR between them."""
+        directory = os.path.join(self.dataBasePath, 'directory1')
+        pathList = wawCommons.getFilesAtPath([directory], ["*.test", "*"])
+
+        testFile1 = os.path.abspath(os.path.join(directory, "file1.test"))
+        assert testFile1 in pathList
+        testFile2 = os.path.abspath(os.path.join(directory, "fileA.ext"))
+        assert testFile2 in pathList
```
Comment (ondrejtomas) [inline ci/unit_tests/wawCommons/getFilesAtPath_test.py]: More tests (suggestions):
- see comment above for inspiration
- test nonExistentFolder
- add more paths to nonExistent* tests to see if it does not skip the rest when finds the first non existing path
- multiple patterns on multiple paths

File: scripts/wawCommons.py [hunk 3]
```diff
@@ -210,24 +210,57 @@ def toEntityName(NAME_POLICY, userReplacements, entityName):
...
+    """
+    Returns generator which yields all files in specified directory (and subdirectories) that match
+    one of the patterns.
+    """
+    for dirpath,_,filenames in os.walk(directory):
+        for f in filenames:
+            if _fileMatchesPatterns(f, patterns):
+                yield os.path.abspath(os.path.join(dirpath, f))
+           
+def _fileMatchesPatterns(filename, patterns):
```
[the remark this answers is not in the document]
Comment (ota-sandr) [reply scripts/wawCommons.py]: I did not know if `absoluteFilePaths` function is used (or planned to be used) somewhere else in WAW. But file `_fileMatchesPatterns` was created in this branch by me, thus I know that this is only place where the function is used. It's an open question if we want to test private functions, too.

File: ci/unit_tests/wawCommons/absoluteFilePaths_test.py (hunk 1 above)
Comment (ota-sandr) [reply ci/unit_tests/wawCommons/absoluteFilePaths_test.py]: Tests were added

File: ci/unit_tests/wawCommons/getFilesAtPath_test.py (hunk 2 above)
Comment (ota-sandr) [reply ci/unit_tests/wawCommons/getFilesAtPath_test.py]: Tests were added

File: scripts/wawCommons.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ota-sandr) [reply scripts/wawCommons.py]: We decided not to test the private functions now.

**Pull request merged**